### PR TITLE
Activity and discussion attachments

### PR DESCRIPTION
**Pull request by lautis**
Add documentation to upload attachments as part of integration messages.

### CODE REVIEW

File: docs/Files.md [hunk 1]
```diff
@@ -1,34 +1,37 @@
...
+You can [upload individual files using the Messages resource](messages#/send/files) of the REST API. Here's a quick `curl` example:
 
-```
-curl -v -X POST -F "event=file" -F "content=@path/to/file.png" http://BASIC-AUTH@api.flowdock.com/flows/ORGANIZATION/FLOW/messages
-```
+    curl -v -X POST -F "event=file" -F "content=@path/to/file.png" http://BASIC-AUTH@api.flowdock.com/flows/ORGANIZATION/FLOW/messages
+
+Alternatively, as an integration developer, files can be uploaded as attachments to `activity` and `discussion` messages.
+
+    curl -H "Accept: application/json" -F "attachments[id]=@localfile.jpg" -F "author[name]=Uploaded" -F "event=activity" -F 'title="Upload a file"' -F 'body=test <img src="cid:cid">' -F "thread[title]='Thread title'", -F "external_thread_id=asdfasdf" -F "flow_token=FLOW_TOKEN" http://api.flowdock.com/messages
```
Comment (vsaarinen) [inline docs/Files.md]: Is `img src="cid:cid">` correct?

Comment (vsaarinen) [inline docs/Files.md]: Is the id of the file in this case "id"?

File: docs/Files.md [hunk 2]
```diff
@@ -1,34 +1,37 @@
...
+
+Note that here the uploaded file is given an id in attributes and referenced by that name in body with `cid:id`. This is optional, but either all attachments must have an id or it should be omitted for all.
```
Comment (vsaarinen) [inline docs/Files.md]: So specifying an id is optional, or referencing it in the body?

File: docs/Files.md (hunk 1 above)
Comment (lautis) [inline docs/Files.md]: Yes, should be `cid:id`. Will fix.

File: docs/Files.md (hunk 2 above)
Comment (lautis) [inline docs/Files.md]: Both are, but you need to specify the uuid to be able to reliably reference an attachment in the body.

**Pull request merged**